### ISSUE TITLE
Error in GatedGraphConv when out_feats is smaller than node feature dimension

### ISSUE THREAD

**Issue by MrGreyfun** [dmlc/dgl#6042]
## 🐛 Bug
When using the `GatedGraphConv`, an error is raised if the `out_feats` parameter is smaller than the dimension of the node feature vectors. This error occurs during the zero padding generation step in the `forward` method of the `GatedGraphConv` class. 

Here is the relevant source code from `gatedgraphconv.py`:

```python
zero_pad = feat.new_zeros(
    (feat.shape[0], self._out_feats - feat.shape[1])
)
```
In this code snippet, `self._out_feats` represents the dimension of the output feature vectors, and `feat` is a matrix consisting of node feature vectors. The expression `feat.shape[1]` returns the dimension of the node feature vectors. The expression `self._out_feats - feat.shape[1]` appears to calculate the dimension of the zero padding vector. However, if the dimension of the output feature is smaller than the node feature dimension, it will result in a negative number, leading to an error when trying to create a tensor with a negative dimension.

I'm unsure if this is a bug or if the gated graph convolution network is intentionally designed to not allow a node feature dimension greater than the output feature dimension. If it is intentional, I suggest adding this information to the documentation and using an assertion error with an explanatory message instead of the current implementation in the source code.
  
## To Reproduce

Run the following code:
```python
import dgl
import torch as th
from dgl.nn import GatedGraphConv

g = dgl.graph(([0,1,2,3,2,5], [1,2,3,4,0,3]))
feat = th.ones(6, 10)
conv = GatedGraphConv(10, 5, 2, 1)
res = conv(g, feat)
```

## Environment

 - DGL Version : 1.1.0+cu113 
 - Backend Library & Version : PyTorch 1.12.0+cu113
 - OS : Linux
 - How you installed DGL : pip 
 - Python version: 3.9

**Comment by czkkkkkk** [comment]
@paoxiaode Could you take a look at this issue?

**Comment by frozenbugs** [comment]
@paoxiaode Is there any update?

**Comment by paoxiaode** [comment]
I read the original paper(https://arxiv.org/pdf/1511.05493.pdf) and the code here (https://github.com/chingyaoc/ggnn.pytorch/blob/0c7897fe9b05e9b4f9a963ff55bd3ad917ea734e/model.py#L69)

Generally speaking, in GatedGraphConv, the output dim(state dimension) must be no less than input dimension(annotation dimension)

We can **raise a ValueError** when input dimension is greater than output dimension
@czkkkkkk @frozenbugs

**Comment by czkkkkkk** [comment]
> I read the original paper(https://arxiv.org/pdf/1511.05493.pdf) and the code here (https://github.com/chingyaoc/ggnn.pytorch/blob/0c7897fe9b05e9b4f9a963ff55bd3ad917ea734e/model.py#L69)
> 
> Generally speaking, in GatedGraphConv, the output dim(state dimension) must be no less than input dimension(annotation dimension)
> 
> We can **raise a ValueError** when input dimension is greater than output dimension @czkkkkkk @frozenbugs

Thanks! Could you make a PR to fix it?

**Comment by MrGreyfun** [comment]
Thank you for taking you time to review and response my issue!
